### PR TITLE
use new sharded optim v2 api

### PR DESCRIPTION
**Pull request by ver217**

### CODE REVIEW

File: zero/colossalai_utils/utils.py [hunk 1]
```diff
@@ -26,40 +29,46 @@ def init_w_col(builder):
...
     else:
+        model = build_model()
         model = ShardedModel(model, **gpc.config.zero)
 
     criterion = build_loss()
 
     print_log(f'Peak Memory = {max_memory_allocated(rank) / (1024 * 1024)} M')
     reset_peak_memory_stats(rank)
 
+    optimizer_cls = build_optimizer
```
Comment (feifeibear) [inline zero/colossalai_utils/utils.py]: name change : build_optimizer -> optimizer_class

**Review by FrankLeeeee — APPROVED** [review]

**Comment by FrankLeeeee** [comment]
LGTM.

File: zero/colossalai_utils/utils.py [hunk 2]
```diff
@@ -26,40 +29,46 @@ def init_w_col(builder):
...
-    #                      shard_param=True):
-    model = build_model()
 
+    print_log('Building model')
     if use_v2:
+        # shard_strategy = TensorShardStrategy()
+        shard_strategy = BucketTensorShardStrategy()
+        with ZeroInitContext(convert_fp16='fp16' in gpc.config,
+                             #  target_device=torch.device(
+                             #      gpc.config.zero.offload_config.device),
```
Comment (feifeibear) [inline zero/colossalai_utils/utils.py]: rm comments

Comment (ver217) [reply zero/colossalai_utils/utils.py]: DONE

File: zero/colossalai_utils/utils.py (hunk 1 above)
Comment (ver217) [reply zero/colossalai_utils/utils.py]: DONE

**Pull request merged**